### ISSUE TITLE
Agenda

### ISSUE THREAD

**Issue by jni** [numfocus/scipy-2019-funding-foss-bof#4]
I made some notes here:

https://github.com/numfocus/scipy-2019-funding-foss-bof/blob/d4107635bafa3cedfcf8d5d8a83776ff679da24e/notes.markdown

(latest version [here](https://github.com/numfocus/scipy-2019-funding-foss-bof/blob/master/notes.markdown))

that could be used to guide the discussion by the moderators. Feel free to append/edit/ask questions about it.

**Comment by amueller** [comment]
Thanks. I'll be busy for about the next two weeks (alternating between sick in bed and holding a workshop), but I definitely want to follow up later.

**Comment by aterrel** [comment]
“Funding people not projects” is the exact thing I’m not interested in :)

It works okay in practice until the money runs out. The competing interest of a project and an individual pi’s career often introduces code and processes that are unsustainable 

Hehe so maybe that is a good discussion to have.

**Comment by rgommers** [comment]
> “Funding people not projects” is the exact thing I’m not interested in :)

Can't say I'm "not interested" because projects are still driven by one or a few people usually, and you do need PIs for pretty much every big chunk of money. But yes, agree with the sentiment in principle - funders should look to fund projects with impact, under the condition that the people to manage the funds effectively are available.

**Comment by jni** [comment]
@aterrel yes, a discussion is definitely warranted. =)

I think I inadvertently made the wrong point with that line. I don't mean that grants should only go to individuals. Rather, that grants should be doled out in a way that allows individuals to sustain a career in open source contribution. Specifically, it would be difficult and stressful to fund a career on NumFOCUS-style small development grants.

**Comment by rgommers** [comment]
> Specifically, it would be difficult and stressful to fund a career on NumFOCUS-style small development grants.

Note that small development grants were never meant for sustaining a career. I would not put those on the agenda, they're not really relevant in this context.

> Rather, that grants should be doled out in a way that allows individuals to sustain a career in open source contribution.

I would translate that as: we need stable funding mechanisms with a 3-5 year or longer timeline, not 1-2 years. The grants-careers connection is a bit indirect other than that. If we grow projects' income to the scale that they can sustain careers, that's likely to be through a multitude of government and nonprofit grants as well as corporate contributions and other funding mechanisms that are sprouting up everywhere.

**Comment by jni** [comment]
> Note that small development grants were never meant for sustaining a career. I would not put those on the agenda, they're not really relevant in this context.

Sure! In no way did I mean to pick on it. It's just the only "direct funding" model I've seen deployed "at scale". Apologies if I have missed others. Part of the purpose of the BoF is to get everyone on the same page about what is out there vs what we would like to be out there.

> I would translate that as: we need stable funding mechanisms with a 3-5 year or longer timeline, not 1-2 years.

Agreed.

> other funding mechanisms that are sprouting up everywhere.

As a side note, I very much like a form of the "share-in-savings" idea discussed in the NumPy mailing list thread ([Apr](https://mail.python.org/pipermail/numpy-discussion/2019-April/079331.html), [May](https://mail.python.org/pipermail/numpy-discussion/2019-May/079440.html)), as it recognises the inherent value of open source (rather than being structured as a donation or charity). It should be *easy* for researchers to put "open source software support" as a line item in their grant budgets, and have an organisation (such as NumFOCUS) charge them accordingly. I think @aterrel mentioned on Twitter that this was being discussed/implemented at NumFOCUS in the form of a service (or similar?) contract?

**Comment by rgommers** [comment]
> As a side note, I very much like a form of the "share-in-savings" idea

The principle is nice indeed. The actual practice of that isn't it seems, it's based on a very obscure law and very hard to find any info on; I'm following up on it with NASA but I doubt it's going to be useful.

> It should be easy for researchers to put "open source software support" as a line item in their grant budgets

In my experience, decent sized grants always have a reasonable amount of flexibility in how they get spent. You don't have to have line items in a budget for every piece of equipment either (things break, new insights, etc.). So having say 0.5% of a budget go to open source, requires probably no more than:
1. Us asking for it clearly in a structured way (which we have never done). That includes better roadmaps etc.
2. Us providing paperwork like a services agreement. NumFOCUS can do this.
3. Socializing the idea.

**Comment by jni** [comment]
@rgommers 

> So having say 0.5% of a budget go to open source, requires probably no more than

I totally agree, and thank you for structuring the steps required. For smaller FOSS projects with less cumulative experience (such as scikit-image), help from e.g. NumFOCUS on steps 1 and 2 would be essential.

> decent sized grants always have a reasonable amount of flexibility in how they get spent

This is somewhat true, but in Australia at least, budgets requested have to be justified in a fair amount of detail, and I can imagine that "paying for something perceived to be free" would often be struck down. Universities here might also put up some barriers. So I think step (3) is critically important.

In other news, I presume everyone on this thread is now aware of [CZI's new FOSS grants](https://chanzuckerberg.com/rfa/essential-open-source-software-for-science/), which should be discussed in the BoF.

**Comment by rgommers** [comment]
> I totally agree, and thank you for structuring the steps required. For smaller FOSS projects with less cumulative experience (such as scikit-image), help from e.g. NumFOCUS on steps 1 and 2 would be essential.

I think it's developing a model that's essential, transferring it to other projects is way easier.

Paperwork and socializing aren't too hard probably, the _1. Us asking for it clearly in a structured way_ is the essential step. If we can get there by or quickly after SciPy'19, I'll be super happy.

**Comment by rgommers** [comment]
> This is somewhat true, but in Australia at least, budgets requested have to be justified in a fair amount of detail, and I can imagine that "paying for something perceived to be free" would often be struck down. Universities here might also put up some barriers. So I think step (3) is critically important.

Don't you think that this will be a matter of "creative bookkeeping"? Let a PI figure out what goes at their university, perhaps an agreement that contributes funds for a specific feature rather than a generic services agreement. There are rules to be respected (and work around), but if a PI wants to spend funds on something then it's hard to imagine that a university really goes out of its way to block that.

**Comment by jni** [comment]
> There are rules to be respected (and work around),

What I'm trying to say is that the service contract, or license, or whatever it's called, should try to do as much of the heavy lifting here as possible. For example, even though I manage my own travel budget, my university has quite strict travel rules that I must abide by. For example, I am (hypocritically, shamefully) not allowed to purchase carbon offsets for my trips with funds managed by the university. There may or may not be restrictions (not sure) about whether I'm allowed to make charitable donations — I presume I am not. In that case, these service agreements should not be structured as charitable donations. Etc.

Yes, creative bookkeeping will be part of the solution, but let's aim to not require *too much* creativity from PIs, or else they just won't bother.

**Comment by jni** [comment]
Here's a hackmd link for the "Agenda", which is currently a bunch of unstructured notes. Please do not delete ideas/topics/notes at this stage: let's try to add for now, then structure, then trim. This document does not need to be ready my May 31, but rather a week or two before SciPy.

https://hackmd.io/6svmf792T1uajPl6YuFw8Q

**Comment by GenevieveBuckley** [comment]
Adding on from the discussion @jni had about [Tidelift](https://thenewstack.io/compensating-open-source-maintainers-tidelifts-on-a-roll/) the other day, the newly launched [GitHub sponsors](https://github.blog/2019-05-23-announcing-github-sponsors-a-new-way-to-contribute-to-open-source/) (which is like Patreon for programmers) ties in with your first bullet point *"fund people not projects?"*

**Comment by rgommers** [comment]
I had a couple of discussions about GitHub Sponsors with them. GitHub understands that the Sponsors feature is limited to individuals right now and is very interested to expand it to projects/orgs. They just need some experience with the feature to figure out exactly how to do it.

**Comment by jni** [comment]
I've added a link in the agenda to @rgommers's [recent blog post](https://labs.quansight.org/blog/2019/05/community-driven-opensource-funded-development/) about community work orders (CWOs).

**Comment by rgommers** [comment]
> This document does not need to be ready my May 31, but rather a week or two before SciPy.

That of course didn't quite work out ....

I think we need to line up a few speakers for the first 20 minutes or so that give a talk of a few minutes about their key idea/initiative. Does that make sense?

- @jni do you know if one of the organizers of the CZI grant program is attending? If not, could you talk about that?
- I can check with the Tidelift folks if any of them will attend.
- GitHub Sponsors, in the unlikely case one of the GitHub folks involved is around (I can check).
- A Sloan/Moore representative, if one happens to be present. Or perhaps someone feels comfortable enough to talk about, e.g., NumFOCUS' relationship with Sloan/Moore or something like that?
- Perhaps lessons/ideas from the recent NSF rejection (@amueller ?)
- Any ideas who could sensibly talk about traditional grants with a fresh/authoritative perspective (or should we leave that out?)?
- I'd personally really like to talk about 3 topics (but that's too much). So in order of priority:

1. Collaboratively formulating our needs and then going out to various funders with a clear request (https://github.com/numfocus/scipy-2019-funding-foss-bof/issues/4#issuecomment-489993294 and the NASA request I already made)
2. The question of what (activities/people) to spend funds on once a project does get funded. _(any takers?)_
3. Quansight's model / Quansight Labs. _(Travis could do this one)_

And topics on the current hackmd I'd prefer to avoid:
- "overhead on a grant" (this is really boring, and everyone will want to bikeshed)
- "savings over buying proprietary software" - I followed up on that with NASA, and it's very esoteric and not that viable
- "bounties" - they don't really work, the social model is all wrong (as evidenced by that they've been around for ages and never made an impact, plus recent experience with IBM offering a bounty for NumPy work)

**Comment by jni** [comment]
> I think we need to line up a few speakers for the first 20 minutes or so that give a talk of a few minutes about their key idea/initiative. Does that make sense?

Yes, although I worry that you are trying to throw too much in here, and I might suggest that instead you request brief snippets of introduction from each of the named orgs/programs and just have you and @lsilen give the 1-2 sentence introduction for each program and the person present representing that program (if any). Otherwise I expect it might last a lot longer than 20min.

However if you think you can effectively shepherd all the people/orgs you listed into a 20 min slot, then go for it!

> do you know if one of the organizers of the CZI grant program is attending? If not, could you talk about that?

Yes, both @freeman-lab and @sofroniewn will be there. They are actually giving a full talk about it (check the schedule), so a brief description is probably sufficient here.

> Perhaps lessons/ideas from the recent NSF rejection (@amueller ?)

I think @amueller already agreed to say a few words about this ([ref](https://github.com/numfocus/scipy-2019-funding-foss-bof/issues/2#issuecomment-494424621)).

> The question of what (activities/people) to spend funds on once a project does get funded. (any takers?)

I feel like someone from NumPy or Jupyter would be the right fit here. @stefanv is the obvious choice for me, since he is managing the NumPy grant right now.

> And topics on the current hackmd I'd prefer to avoid:

Yep, those are fine, as mentioned the notes are for discussion in the preparation stage. Feel free to move them to a "rejected ideas" section with a brief note or link to your comment. Regarding overheads, I didn't really mean that there should be a discussion, only to note that they are important and any grants that explicitly forbid university overheads are not that valuable in academia.

**Comment by amueller** [comment]
I think it would be great to get someone from Jupyter to talk, maybe Brian has time?

And yes, happy to talk about the NSF grant and the rest of my experience with NSF and  the discussions I had since.

**Comment by rgommers** [comment]
> Yes, although I worry that you are trying to throw too much in here, and I might suggest that instead you request brief snippets of introduction from each of the named orgs/programs and just have you and @lsilen give the 1-2 sentence introduction for each program and the person present representing that program (if any). Otherwise I expect it might last a lot longer than 20min.

Yeah it's going to be a challenge. My expectation though is that we don't do it and go straight to a discussion, people will bring up these topics anyway and then we have to go through it in a less planned fashion and it will take longer.

CZI has a full talk, that's great. In that case 1-2 sentences here is enough, and we refer to that talk. Also, we have a NumPy talk so can share NumPy experiences there. That's two down.

I'll email Brian to see if he can talk about Jupyter.

**Comment by lsilen** [comment]
Keeping it to a brief introduction of 2-3 min should leave time for discussion. I agree with @jni though, that might be difficult.

**Comment by sofroniewn** [comment]
@rgommers - Nick here from CZI. Jeremy @freemanlab and I will both be at the BoF - which I think is Thursday 1pm. Unfortunately that will be before our talk (at 2:35pm same day), but Jeremy and I are happy to provide info on the new [open source grant program](https://chanzuckerberg.com/rfa/essential-open-source-software-for-science/) and answer any questions

**Comment by rgommers** [comment]
Thanks @sofroniewn. If your talk is after, I think it would be useful to still give a 1-2 minute intro, e.g. what is the call about, what motivated it, and saying you'll give a full talk later in the day.

Brian confirmed he can talk about Jupyter, sharing Jupyter's key lessons/experiences. 

> Keeping it to a brief introduction of 2-3 min should leave time for discussion. I agree with @jni though, that might be difficult.

I'm happy to police that strictly. If people can do it for lightning talks, they should be able to do it here if we make it clear beforehand that timing will be strict.

**Comment by rgommers** [comment]
@sofroniewn and @amueller, if you want to use some slides to support your 2-3 min lightning talk, could you please get them to me by Tuesday evening in pdf format? I'll put them all together so we don't lose time with laptop switches.
